browser/ui/webui/ai_chat/ai_chat_ui.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/ai_chat/ai_chat_ui.h"
+
+#include <utility>
+
+#include "brave/browser/ai_chat/constants.h"
+#include "brave/browser/ai_chat/pref_names.h"
+#include "brave/browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h"
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/components/ai_chat_ui/resources/page/grit/ai_chat_ui_generated_map.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/l10n/common/localization_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/common/url_constants.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+
+AIChatUI::AIChatUI(content::WebUI* web_ui)
+    : ui::UntrustedWebUIController(web_ui),
+      profile_(Profile::FromWebUI(web_ui)) {
+  DCHECK(profile_);
+  browser_ = chrome::FindBrowserWithProfile(profile_);
+
+  // Create a URLDataSource and add resources.
+  content::WebUIDataSource* untrusted_source =
+      content::WebUIDataSource::CreateAndAdd(
+          web_ui->GetWebContents()->GetBrowserContext(), kChatUIURL);
+
+  webui::SetupWebUIDataSource(
+      untrusted_source,
+      base::make_span(kAiChatUiGenerated, kAiChatUiGeneratedSize),
+      IDR_CHAT_UI_HTML);
+
+  untrusted_source->AddResourcePath("styles.css", IDR_CHAT_UI_CSS);
+
+  for (const auto& str : ai_chat::kLocalizedStrings) {
+    std::u16string l10n_str =
+        brave_l10n::GetLocalizedResourceUTF16String(str.id);
+    untrusted_source->AddString(str.name, l10n_str);
+  }
+
+  untrusted_source->AddBoolean(
+      "hasSeenAgreement",
+      profile_->GetOriginalProfile()->GetPrefs()->GetBoolean(
+          ai_chat::prefs::kBraveChatHasSeenDisclaimer));
+
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      "script-src 'self' chrome-untrusted://resources;");
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::StyleSrc,
+      "style-src 'self' 'unsafe-inline' chrome-untrusted://resources;");
```

Comment:
Do we still need `unsafe-inline`? I thought the inline styles had been removed.

Reply:
Afraid all our WebUIs need unsafe-inline because that's how webpack injects styles to the page (and also styled-components). We can reconfigure this but I'd like to do it for all WebUIs in a follow-up. Most of our WebUIs don't define a style-src or a default-src policy so unsafe-inline is allowed. https://github.com/brave/brave-browser/issues/29431